wp-content/plugins/hurumap/api.php

```diff
@@ -90,204 +44,89 @@ function register_routes()
+        $charts[] = $chart;
     }
-    $response = new WP_REST_Response($res);
-    $response->set_status(200);
-}
 
-function update_or_create_sections($request)
-{
-    $json = $request->get_json_params();
-    if (is_array($json[0])) {
-        foreach ($json as $section) {
-            update_or_create_section($section);
-        }
-    } else {
-        update_or_create_section($json);
+    $posts = get_posts(array(
+        'posts_per_page'			=> -1,
+        'post_type'					=> 'hurumap-section',
+        'post_status'				=> array('publish')
+    ));
+    $sections = array();
+    foreach( $posts as $post ) {
+        $section = json_decode($post->post_content);
+        $section->charts = array_filter($charts, function($a) use ($section) {
+            return $a['section'] == $section->id;
+        });
+        $sections[] = $section;
     }
-    $response = new WP_REST_Response();
-    $response->set_status(200);
-}
 
-function delete_sections($request)
-{
-    global $wpdb;
+    usort($sections, function($a, $b) {
+        return $a->order > $b->order;
+    });
 
-    $json = $request->get_json_params();
-    $placeholders = implode(', ', array_fill(0, count($json), '%s'));
-    $success = $wpdb->query($wpdb->prepare("DELETE FROM {$wpdb->base_prefix}chart_sections WHERE id IN ({$placeholders})", $json));
-    $response = new WP_REST_Response($success);
-    $response->set_status(200);
+    $response = new WP_REST_Response($sections);
 
-    return $response;
-}
-
-function update_or_create_flourish_chart($json)
-{
-    global $wpdb;
-    var_dump($json);
-    if (!$wpdb->update(
-        "{$wpdb->base_prefix}flourish_charts",
-        $json,
-        array('id' => $json['id'])
-    )) {
-        $wpdb->insert("{$wpdb->base_prefix}flourish_charts", $json);
-    }
-}
-function update_or_create_flourish_charts($request)
-{
-    $json = $request->get_json_params();
-    if (is_array($json[0])) {
-        foreach ($json as $chart) {
-            update_or_create_flourish_chart($chart);
-        }
-    } else {
-        update_or_create_flourish_chart($json);
-    }
-    $response = new WP_REST_Response();
     $response->set_status(200);
-    return $response;
-}
-
 
-function delete_flourish_charts($request)
-{
-    global $wpdb;
-    $json = $request->get_json_params();
-    $success = $wpdb->delete($wpdb->base_prefix . "flourish_charts", array('id' => $json['id']));
-    $response = new WP_REST_Response($success);
-    $response->set_status(200);
     return $response;
 }
 
 function get_flourish_chart($request)
 {
-    global $wpdb;
-
     WP_Filesystem();
 
-    $chart_id = $request->get_param('chart_id');
-    $flourish = $wpdb->get_results($wpdb->prepare("SELECT * FROM {$wpdb->base_prefix}flourish_charts where id=%s  LIMIT 1", $chart_id));
-    $file_content = $flourish[0]->file;
+    $id = $request->get_param('chart_id');
+    $post = get_post($id);
+    if ($post) {
+        $content = json_encode($post->post_content);
```

Comment:
Again need to at least check if it's `hurumap-visual` post (and it's actually of type `flourish`)